### ISSUE TITLE
--build ignoring tsconfig.json changes

### ISSUE THREAD

**Issue by asmundg** [Microsoft/TypeScript#26503]
**TypeScript Version:**  3.1.0-dev.20180817

**Search Terms:** --build tsconfig

**Code**

In [learn-a](https://github.com/RyanCavanaugh/learn-a), modifying package tsconfig.json files does not trigger rebuild when the package has already been built once.

```shell
yarn tsc --build packages/pkg1 --verbose
sed -i 's/"rootDir": "src"/"rootDir": "src","target":"es3"/' packages/pkg1/tsconfig.json
yarn tsc --build packages/pkg1 --verbose
```

**Expected behavior:**

pkg1 should rebuild after updating tsconfig.json

**Actual behavior:**

pkg1 is not rebuilt.

First execution of tsc:
```
message TS6355: Projects in this build: 
    * packages/pkg1/tsconfig.json

message TS6352: Project 'packages/pkg1/tsconfig.json' is out of date because output file 'packages/pkg1/lib/foo.js' does not exist

message TS6358: Building project '/home/asgramme/src/learn-a/packages/pkg1/tsconfig.json'...
```

Second execution of tsc, after modifying tsconfig.json:
```
message TS6355: Projects in this build: 
    * packages/pkg1/tsconfig.json

message TS6351: Project 'packages/pkg1/tsconfig.json' is up to date because newest input 'packages/pkg1/src/foo.ts' is older than oldest output 'packages/pkg1/lib/foo.js'

Done in 0.25s
```

**Related Issues:**

**Comment by AlCalzone** [comment]
I just ran into this aswell. Running `tsc -b --clean` before `tsc -b` sort of "fixes" it.
